### PR TITLE
Update TableViewCell sublabel accessory view margin

### PR DESCRIPTION
**Pull request by laminesm**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

There has been a design request to change the padding between the icons and sublabel of the `TableViewCell` from 8pt to 4pt. Currently, both the main label and the sublabel use the same 8pt `labelAccessoryViewMarginTrailing` for the padding. I've added a `sublabelAccessoryViewMarginTrailing` token to be used for the sublabel. The `layoutLabelViews` function now takes in the appropriate padding token to be applied to the labels. 

A new fluent icon was added as well to test the change properly as the icon it replaces was not representative of the icons used on the design figmas.

NOTE: I've run the numbers multiple times and can't figure out why libFluentUI is registering a decrease in binary size even though the individual files have had binary size increases. 

Binary change:

| File | Before | After | Delta |
|------|--------|-------|-------|
| libFluentUI.a | 29,106,328 bytes | 29,105,816 bytes | - 512 bytes |
| TableViewCell | 838,400 bytes | 840,016 bytes | 1,616 bytes |
| TableViewCellTokenSet | 112,576 bytes | 113,384 bytes | 808 bytes |

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before | Screenshot 2023-01-20 at 2 45 40 PM |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1500)

### CODE REVIEW

File: ios/FluentUI.Demo/FluentUI.Demo/Resources/Assets.xcassets/TableViewCell/ic_fluent_lock_closed_20_regular.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "filename" : "ic_fluent_lock_closed_20_regular.pdf",
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/Resources/Assets.xcassets/TableViewCell/ic_fluent_lock_closed_20_regular.imageset/Contents.json:4]: hmmm i would have expected there will be one of this. what is the asset setting on the image? single scale?

Comment (harrieshin) [reply ios/FluentUI.Demo/FluentUI.Demo/Resources/Assets.xcassets/TableViewCell/ic_fluent_lock_closed_20_regular.imageset/Contents.json:4]: like there should n't be 1x, 2, 3x in the json file

File: ios/FluentUI.Demo/FluentUI.Demo/TableViewCellSampleData.swift [hunk 2]
```diff
@@ -27,15 +27,15 @@ class TableViewCellSampleData: TableViewSampleData {
                 Item(text1: "Contoso Survey",
                      text2: "Research Notes",
                      image: "excelIcon",
-                     text2LeadingAccessoryView: { createIconsAccessoryView(images: ["shared-12x12", "success-12x12"]) })
+                     text2LeadingAccessoryView: { createIconsAccessoryView(images: ["shared-12x12", "ic_fluent_lock_closed_20_regular"]) })
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/TableViewCellSampleData.swift:30]: can we update the shared-12x12 too then? so it looks uniform and fluent2 style?

File: ios/FluentUI/Table View/TableViewCellTokenSet.swift [hunk 3]
```diff
@@ -213,9 +213,12 @@ extension TableViewCellTokenSet {
     /// The leading margin for the labelAccessoryView.
     static let labelAccessoryViewMarginLeading: CGFloat = GlobalTokens.spacing(.size80)
 
-    /// The trailing margin for the labelAccessoryView.
+    /// The trailing margin for the labelAccessoryView of the title label.
     static let labelAccessoryViewMarginTrailing: CGFloat = GlobalTokens.spacing(.size80)
 
+    /// The trailing margin for the labelAccessoryView of the subtitle label.
+    static let sublabelAccessoryViewMarginTrailing: CGFloat = GlobalTokens.spacing(.size40)
```
Comment (harrieshin) [inline ios/FluentUI/Table View/TableViewCellTokenSet.swift:220]: i think we should match the name of the actual UILabel. ex) subtitleLabelAccessoryViewMarginTraining and titleLableAccessoryviewMarginTrailing for easy search. which is what subtitletwolineheight did as well.

File: ios/FluentUI.Demo/FluentUI.Demo/Resources/Assets.xcassets/TableViewCell/ic_fluent_lock_closed_20_regular.imageset/Contents.json (hunk 1 above)
Comment (laminesm) [reply ios/FluentUI.Demo/FluentUI.Demo/Resources/Assets.xcassets/TableViewCell/ic_fluent_lock_closed_20_regular.imageset/Contents.json:4]: Yep, I'll remove them.

**Review by harrieshin — APPROVED** [review]

**Pull request merged**